File: 02/two.py
#!/bin/env python3

location = {
    'position': 0,
    'depth': 0,
    'aim': 0
}
dirs = {
    'forward': (
        ('position', int),
        ('depth', lambda x: int(x) * location['aim'])
    ),
    'down': (('aim', int),),
    'up': (('aim', lambda x: -int(x)),)
}

with open('input.txt') as infile:
    for line in infile:
        direction, distance = line.split()
        for axis, parse in dirs[direction]:
            location[axis] += parse(distance)

print(location['position'] * location['depth'])

File: 02/two.nim
import std/strutils

var
  position = 0
  depth = 0
  aim = 0
  direction: string
  distance: string
  idistance: int

for line in "input.txt".lines:
  (direction, distance) = line.split
  idistance = distance.parse_int
  case direction
  of "forward":
    position.inc idistance
    depth.inc idistance * aim
  of "down":
    aim.inc idistance
  of "up":
    aim.dec idistance

(position * depth).echo
